### PR TITLE
fix: Migrate CircleCI to GHA

### LINKED ISSUE #408: Unblock or replace CircleCI jobs
CircleCI jobs have stopped running. Test/build/release jobs in CircleCI should be either unblocked or replaced with GitHub Actions.

### LINKED ISSUE #411: Convert CircleCI automation to GitHub Actions
This repo uses both, and CircleCI jobs have not been reliably run for reasons that aren't clear.

### PR DESCRIPTION
**Pull request by mrickard**
Added GitHub Actions workflows for validate (lint, test), and publish.

Closes #411 
Closes #408 
Closes NR-248539

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        node-version: [lts/*]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3 # https://github.com/actions/setup-node
+        with:
+          node-version: ${{ matrix.node-version }}
+          registry-url: 'https://registry.npmjs.org'
+      - run: cp .npmrc-ci .npmrc
```
Comment (bizob2828) [inline .github/workflows/publish.yml:21]: This line isn't need and subsequently can remove this file

File: .github/workflows/validate.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: Validate
+
+on: [push, workflow_dispatch]
```
Comment (bizob2828) [inline .github/workflows/validate.yml:3]: Is this the typical CI file? If so, shouldn't this have pull_request on it?

File: .github/workflows/validate.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install Dependencies
+        run: npm ci
+      - name: Build
+        run: npm run build
+      - name: Run Snapshot Tests
+        run: npm test
```
Comment (bizob2828) [inline .github/workflows/validate.yml:43]: I don't see this uploading covering like it was doing in circle ci

**Review by bizob2828 — CHANGES_REQUESTED** [review]

File: .github/workflows/validate.yml (hunk 2 above)
Comment (mrickard) [reply .github/workflows/validate.yml:3]: This one started with the workflow for repolinter in this repo, which isn't triggered by PRs. I'll add it.

File: .github/workflows/validate.yml (hunk 3 above)
Comment (mrickard) [reply .github/workflows/validate.yml:43]: Ah, thank you! It's uploading now.

File: .github/workflows/validate.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Archive Unit Test Coverage
+        uses: actions/upload-artifact@v3
```
Comment (bizob2828) [inline .github/workflows/validate.yml:45]: this is just creating a zip it is not posting to codecov

**Review by bizob2828 — CHANGES_REQUESTED** [review]

File: .github/workflows/validate.yml (hunk 4 above)
Comment (mrickard) [reply .github/workflows/validate.yml:45]: Aha. I can port in from what we do in the agent.

Comment (bizob2828) [reply .github/workflows/validate.yml:45]: sounds good. it looks like the legacy circle CI plan used a similar action like we do to upload the coverage

Comment (mrickard) [reply .github/workflows/validate.yml]: Looks like we're getting uploads now.

**Review by bizob2828 — APPROVED** [review]
Nice work. looks like you may have to update your required status checks though before merging

**Comment by mrickard** [comment]
@bizob2828 Thank you! Yes--I have to remove the CircleCI checks, and then ensure the GHA checks after I merge.

**Pull request merged**